packages/synapse-react-client/src/components/RecentPublicationsGrid/RecentPublicationsGrid.tsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react'
+import {
+  Grid,
+  Card,
+  CardContent,
+  Typography,
+  Button,
+  Box,
+  useMediaQuery,
+  useTheme,
+} from '@mui/material'
+import { SynapseConstants, SynapseUtilityFunctions } from '../../utils'
+import { QueryBundleRequest } from '@sage-bionetworks/synapse-types'
+import { getFieldIndex } from '../../utils/functions/queryUtils'
+import useGetQueryResultBundle from '../../synapse-queries/entity/useGetQueryResultBundle'
+import dayjs from 'dayjs'
+import { formatDate } from '../../utils/functions/DateFormatter'
+
+export type RecentPublicationsGridProps = {
+  sql: string
+  buttonLink?: string
+  buttonLinkText?: string
+  summaryText?: string
+}
+
+function RecentPublicationsGrid(props: RecentPublicationsGridProps) {
+  const { sql, buttonLink, buttonLinkText, summaryText } = props
+  const theme = useTheme()
+  const isSmallScreen = useMediaQuery(theme.breakpoints.down('sm'))
+  const isMediumScreen = useMediaQuery(theme.breakpoints.between('sm', 'md'))
+  const entityId = SynapseUtilityFunctions.parseEntityIdFromSqlStatement(sql)
+  const queryBundleRequest: QueryBundleRequest = {
+    partMask:
+      SynapseConstants.BUNDLE_MASK_QUERY_SELECT_COLUMNS |
+      SynapseConstants.BUNDLE_MASK_QUERY_RESULTS,
+    concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+    entityId,
+    query: {
+      sql,
+    },
+  }
+  const { data: queryResultBundle } =
+    useGetQueryResultBundle(queryBundleRequest)
+
+  const dataRows = queryResultBundle?.queryResult!.queryResults.rows ?? []
+
+  enum ExpectedColumns {
+    TAG = 'Tag',
+    JOURNAL = 'Journal',
+    TITLE = 'Title',
+    PUBLICATION_DATE = 'publicationDate',
+  }
+
+  const tagColIndex = getFieldIndex(ExpectedColumns.TAG, queryResultBundle)
+  const journalColIndex = getFieldIndex(
+    ExpectedColumns.JOURNAL,
+    queryResultBundle,
+  )
+  const titleColIndex = getFieldIndex(ExpectedColumns.TITLE, queryResultBundle)
+  const publicationDateColIndex = getFieldIndex(
+    ExpectedColumns.PUBLICATION_DATE,
+    queryResultBundle,
+  )
+
+  return (
+    <Box
```

Comment:
What does this component look like when data is loading?

Reply:
I added a spinner when the data is loading!